FormalConjectures/ErdosProblems/40.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+import FormalConjectures.ErdosProblems.«28»
+
+/-!
+# Erdős Problem 40
+
+*Reference:* [erdosproblems.com/40](https://www.erdosproblems.com/40)
+-/
+
+open Filter Set
+open scoped Pointwise
+
+
+/--
+The predicate for a function $g\colon\mathbb{N} → \mathbb{R})$ that
+$$\lvert A\cap \{1,\ldots,N\}\rvert \gg \frac{N^{1/2}}{g(N)}$$
+implies $\limsup 1_A\ast 1_A(n)=\infty$.
+-/
+def Erdos40For (g : ℕ → ℝ) : Prop :=
+  ∀ (A : Set ℕ),
+  ((fun (N : ℕ) => (N : ℝ).sqrt/(g N)) =O[atTop] (fun (N : ℕ) => ((A ∩ Set.Icc 1 N).ncard : ℝ))) →
+  (limsup (fun (N : ℕ) =>
+  letI a := PowerSeries.mk (indicator A 1)
+  (a * a).coeff ℕ N) atTop = (⊤ : ℕ∞))
```

Comment:
We seem to be using this power series trick quite a bit - maybe it would be worth adding something in ForMathlib about that at some point?